### ISSUE TITLE
Class<NSCopying> cast to id<NSCopying> issue in FLEXObjectExplorerFactory

### ISSUE THREAD

**Issue by matrush** [FLEXTool/FLEX#488]
`Class<NSCopying>` cast to `id<NSCopying>` issue in `FLEXObjectExplorerFactory`.

This is probably one of the new diagnostics that are introduced in `clang10`, so for normal Xcode builds it works fine for now. But as this code is not future-proof, we should fix it.

```
FLEX/src/Classes/ObjectExplorers/FLEXObjectExplorerFactory.m:40:13: error: passing 'Class<NSCopying>' to parameter of incompatible type 'id<NSCopying>  _Nonnull const __unsafe_unretained' [-Werror]
            MetaclassKey(NSObject)     : [FLEXClassShortcuts class],
            ^~~~~~~~~~~~~~~~~~~~~~
FLEX/src/Classes/ObjectExplorers/FLEXObjectExplorerFactory.m:41:13: error: passing 'Class<NSCopying>' to parameter of incompatible type 'id<NSCopying>  _Nonnull const __unsafe_unretained' [-Werror]
            ClassKey(NSArray)          : [FLEXCollectionContentSection class],
            ^~~~~~~~~~~~~~~~~
FLEX/src/Classes/ObjectExplorers/FLEXObjectExplorerFactory.m:42:13: error: passing 'Class<NSCopying>' to parameter of incompatible type 'id<NSCopying>  _Nonnull const __unsafe_unretained' [-Werror]
            ClassKey(NSSet)            : [FLEXCollectionContentSection class],
```

The issue is the `Class` and `id` have different definitions:
```
/// An opaque type that represents an Objective-C class.
typedef struct objc_class *Class;
```
```
/// A pointer to an instance of a class.
typedef struct objc_object *id;
```

Therefore the error is reported.

I have attempted to fix this [here](https://github.com/FLEXTool/FLEX/commit/d77864494d5090d9fd2e435e9f04e244da0d10eb) by using `NSString` as the key but seems it's breaking things. So I would like to hear some other solutions.

**Comment by NSExceptional** [comment]
I think the warning is because currently it is written as `Class<NSCopying>` and it expects `id<NSCopying>`. Wouldn't changing the cast to `id<NSCopying>` fix the warning?

**Comment by matrush** [comment]
@NSExceptional Yeah in practice it can, but it doesn't quite make sense as the `Class` and `id` does not always have the same structure. Thoughts?

**Comment by NSExceptional** [comment]
They do, though. `Class` is a supertype of `id`, semantically. They both have a single `Class isa` field.

The compiler is only complaining because they're not strictly the same type, but we know the types are compatible, so we are allowed to cast from `Class` to `id`

**Comment by matrush** [comment]
Cool I found similar answer [here](https://stackoverflow.com/questions/20874458/ios-objective-c-struct-objc-class-and-struct-objc-object). I will patch this by using `id<NSCopying>` then.

**Comment by NSExceptional** [comment]
Alright, thanks! You should only need to modify the macro definitions :)

**Comment by matrush** [comment]
https://github.com/FLEXTool/FLEX/commit/637074b3541d334d5c65b048055540f36f6d6c15 Fixed here

**Comment by NSExceptional** [comment]
Thanks 😄

Random GitHub tip: if you put "Fix #488" in the commit message it'll automatically close the associated issue when you push 😎

**Comment by matrush** [comment]
@NSExceptional Thanks for the pro-tip 😀
